### DISCUSSION TITLE
How to config WS Tunnel in order to use it as a Relay server on Hiddify? |  چگونه می‌توان سرور ریلی با روش WS Tunnel ایجاد کرد و از آن در هیدیفای استفاده نمود؟

### DISCUSSION THREAD

**Discussion by lymanjre** [hiddify/Hiddify-Manager#851] [Q&A سوالات رایج] [answered]
How to config WS Tunnel in order to use it as a Relay server on Hiddify?
چگونه می‌توان سرور ریلی با روش WS Tunnel ایجاد کرد و از آن در هیدیفای  استفاده نمود؟

**Comment by lymanjre** [comment, marked as answer]
# جهت مشاهده مقاله آپدیت شده در خصوص این روش [اینجا](https://hiddify.com/fa/manager/domain-worker-cdn-and-tunneling/Tutorial-for-setting-up-WST-tunnel-as-relay-server/) را ببینید.

**Comment by nimaus4** [comment]
![Shot 0018](https://user-images.githubusercontent.com/131380072/235377453-bbcbcc3e-c5d8-41f1-b61d-51042a5aed71.png)
inja mimone va rad nemishe???

on 8080 ro taghir ham dadam

**Reply by lymanjre** [reply]
یه تغییر دادم. لطفا مطلب بالا را از اول دنبال کن. یه جابجایی انجام دادم فکر کنم مشکل حل بشه

**Reply by Uncoadv** [reply]
منم دقیقا همین مشکل رو روی سرور اصلیم دارم 
گیر می کنه همینجا در صورتی که فایروال خاموشه و همه چیز رو  هم درست رفتم

**Reply by AliAlmasi** [reply]
> یه تغییر دادم. لطفا مطلب بالا را از اول دنبال کن. یه جابجایی انجام دادم فکر کنم مشکل حل بشه

متاسفانه هنوزم همونه، لطفا یبار دیگه بررسی کنین، مرسی.

**Reply by nimaus4** [reply]
بازم نشد
الان میرم رو سرور خارج کلا این دستورو قبوا نمیکنه
![Uploading Shot 0027.png…]()

**Reply by nimaus4** [reply]
![Shot 0027](https://user-images.githubusercontent.com/131380072/235798306-ad1dcb83-b64c-48b7-9e46-d97a61744d7d.png)

**Reply by lymanjre** [reply]
> ![Shot 0018](https://user-images.githubusercontent.com/131380072/235377453-bbcbcc3e-c5d8-41f1-b61d-51042a5aed71.png) inja mimone va rad nemishe???
> 
> on 8080 ro taghir ham dadam

این درسته. منتظر ارتباط از سمت سرور میانی. این تانل دو طرفه اس. یعنی یک سمت شما کانکشن را فعال میکنی( که مال شما در این مرحله است) سپس در سمت مقابل نیز کامند گفته شده را بزنید تا تانل برقرار بشه.

**Comment by lymanjre** [comment]
عزیزان من تست کردم اوکیه. ببینید وقتی کامند را روی سرور اصلی میزنی منتظر کانکشن از سرور میانی روی پورت ۸۰۸۰ است. کافیه شما روی سرور میانی هم کامندی که گذاشتم را بزنید. تانل برقرار میشه.
بعد سرور میانی منتظر کانکشن روی پورتهای ۴۴۳و۸۰ هست. حالا اگه سرور ریلی را اد کنید و کانکشنهاش را استفاده کنید می بینید که تانل برقرار هست. نگران نباشید. تست شده

همچنین یه نکته دیگه که الان اضافه کردم اینه که میتونید به دامین CDN تون وصلش کنید که اینطوری جلو فیلترشدن سرور اصلی را هم بگیرید.

**Reply by hntcAir** [reply]
NEW incoming connection from .......
 Rejecting connection

**Comment by MahdiAkrami01** [comment]
به نظر من از haproxy استفاده کنید بهتره 
من wstunnel تست کردم اپلود زیر 1 هست . ولی با haproxy تست کردم اپلود مشکلی نداشت

**Reply by Uncoadv** [reply]
اموزشش رو از کجا میشه دید؟

**Reply by sajadzi** [reply]
دوست عزیز زحمت میکشی اموزششو بزارید

**Comment by farshadfaza** [comment]
سلام من زدم تو خود پنل هدیفای اعمال تغییرات میزنم این خطارو میده
![IMG_20230517_051326_817.jpg](https://github.com/hiddify/hiddify-config/assets/25201154/6b090d5c-be52-4f65-84c7-611f7bf57782)

**Comment by NabiKAZ** [comment]
الان امتیاز این پورت فورواردینگ نسبت به یه دستور iptables ساده چیه؟

**Comment by yousefi-p** [comment]
سلام خدمت دوستان گرامی
من تست کردم و اروری نمیده اما ارتباط تو v2ray بر قرار نمیشه. ریزه کاری داره که باید انجام بشه؟

**Reply by Mahanfix** [reply]
میشه بگین باید چه ریز کاری رو انجام بدیم واسه من همه چی اوکی پیش رفت اما موقع پینگ گرفتن پینگ نمیده

**Reply by yousefi-p** [reply]
منم اطلاعی ندارم

**Comment by punkgeek** [comment]
@lymanjre
 در دیتاسنتر های ایرانی، زیرساخت بر روی کانکشن ها محدودیت سرعت گذاشته.
از جایی گه وب سکوت با یک کانکشن با سرور هیدیفای ارتباط برقرار میکنه، باعث میشه که از کل پهنای باند سرور تانل نشه استفاده کرد. پیشنهادی برای ایجاد چندین کانکش بر روی wstunnel دارید؟
